File: arrays/transposeMatrix/transposeMatrix.py
def transposeMatrix(matrix):
    transposedMatrix = []
    for col in range(len(matrix[0])):
        newRow = []
        for row in range(len(matrix)):
            newRow.append(matrix[row][col])

        transposedMatrix.append(newRow)

    return transposedMatrix


print(
    transposeMatrix([[0, 4, 2, 42], [-1, 5, 3, 100], [-2, 6, -2, 30], [-3, 7, -3, -42]])
)

File: arrays/transposeMatrix/transposeMatrix.js
// O(w * h) time,  O(w * h) space

function transposeMatrix(matrix) {
  const result = []
  for(let col = 0; col < matrix[0].length; col++) {
    const newRow = []
    for (let row = 0; row < matrix.length; row++) {
      newRow.push(matrix[row][col])
    }
    result.push(newRow)
  }
  return result;
}

console.log(transposeMatrix([
  [0, 4, 2, 42],
  [-1, 5, 3, 100],
  [-2, 6, -2, 30],
  [-3, 7, -3, -42]
]))
